hera_sim/simulate.py

```diff
@@ -434,4 +435,64 @@ def add_xtalk(self, model='gen_whitenoise_xtalk', bls=None, **kwargs):
                 vis=self.data.data_array[blt_ind, 0, :, pol_ind],
                 xtalk=xtalk
             )
+    
+    def make_data(self, sim_file=None, sim_params=None):
+        """
+        Accept a dictionary or YAML file of simulation parameters and add in
+        all of the desired simulation components to the Simulator object.
+
+        Args:
+            sim_file (str): string providing a hook to a YAML file
+            
+            sim_params (dict): dictionary of simulation parameters.
+                it must take the form {model:params}, where model is
+                a string specifying the simulation component to be added
+                and params is a dictionary providing all the keyword
+                arguments and their desired values.
+        """
+        # make a dictionary whose values point to the various methods
+        # used to add different simulation components
+        SIMULATION_COMPONENTS = { 'noiselike_eor':self.add_eor,
```

Comment:
Do you know if there's a sort of plugin that can search through all the other `hera_sim` modules and find all of the possible simulation components? I'm envisioning having something that goes and looks through, say `eor`, `foregrounds`, etc. and pulls out the different simulation functions, then does this sort of mapping where the name maps to the `Simulator` method.

Reply:
I agree that it's better to generate this mapping automatically, but I currently have no idea how to go about implementing that. I'll read up on plugins to see if I can find something that will do the trick, and I'll implement it if I find something that works.

Reply:
So there's no way we can do this well in the current implementation. You can definitely go through each module and find which functions are defined in it -- but there's no automatic way to tell if those functions correspond to "models", nor if they correspond to a prescribed API. 

I think the "best" way to achieve this is to define each model as a class, which has a metaclass that keeps track of its subclasses (i.e. the "plugins" that adopt that API). This would take a bit of setup outside the scope of this PR though. Let's try get this one accepted, and open an issue for it.